test/RewardsManager.test.js

```diff
@@ -0,0 +1,195 @@
+    const TestToken = await ethers.getContractFactory('TestToken')
+    const MockDandelionVotingContract = await ethers.getContractFactory('MockDandelionVotingContract')
+
+    const signers = await ethers.getSigners()
+    owner = signers[0]
+    daoCreator = await ethers.getImpersonatedSigner(DAO_CREATOR)
+    randomGuy = ethers.Wallet.createRandom().connect(ethers.provider)
+    pntHolder1 = ethers.Wallet.createRandom().connect(ethers.provider)
+    pntHolder2 = ethers.Wallet.createRandom().connect(ethers.provider)
+    pntHolder3 = ethers.Wallet.createRandom().connect(ethers.provider)
+    pntHolder4 = ethers.Wallet.createRandom().connect(ethers.provider)
+    pntHolders = [pntHolder1, pntHolder2, pntHolder3, pntHolder4]
+
+    await Promise.all([...pntHolders, daoCreator, randomGuy].map((_dest) => sendEthers(owner, _dest, '1001')))
+
+    acl = await ethers.getContractAt(AclAbi, ACL_ADDRESS)
+    tokenManager = await ethers.getContractAt(TokenManagerAbi, TOKEN_MANAGER_ADDRESS)
+    pnt = await TestToken.deploy('PNT', 'PNT')
+
+    await Promise.all(pntHolders.map((_holder) => sendPnt(owner, _holder.address, '400000')))
+
+    dandelionVoting = await MockDandelionVotingContract.deploy()
+
+    epochsManager = await upgrades.deployProxy(EpochsManager, [EPOCH_DURATION, 0], {
+      initializer: 'initialize',
+      kind: 'uups'
+    })
+
+    rewardsManager = await upgrades.deployProxy(
+      RewardsManager,
+      [
+        await epochsManager.getAddress(),
+        await dandelionVoting.getAddress(),
+        await pnt.getAddress(),
+        await tokenManager.getAddress(),
+        PNT_MAX_TOTAL_SUPPLY
+      ],
+      {
+        initializer: 'initialize',
+        kind: 'uups'
+      }
+    )
+
+    await missingSteps()
+  })
+
+  it('should deploy correctly', async () => {
+    expect(await rewardsManager.token()).to.eq(await pnt.getAddress())
+    expect(await rewardsManager.tokenManager()).to.eq(await tokenManager.getAddress())
+  })
+
+  it('should be possible to deposit tokens', async () => {
+    const amount = 100n
+    expect(await epochsManager.currentEpoch()).to.be.eq(0)
+    await rewardsManager.grantRole(DEPOSIT_REWARD_ROLE, owner.address)
+    await pnt.approve(await rewardsManager.getAddress(), amount)
+    const pntOwnerBalancePre = await pnt.balanceOf(owner.address)
+    const pntRewardsManagerBalancePre = await pnt.balanceOf(await rewardsManager.getAddress())
+    await rewardsManager.depositForEpoch(0, amount)
+    const pntOwnerBalancePost = await pnt.balanceOf(owner.address)
+    const pntRewardsManagerBalancePost = await pnt.balanceOf(await rewardsManager.getAddress())
+    expect(pntOwnerBalancePost).to.be.eq(pntOwnerBalancePre - amount)
+    expect(pntRewardsManagerBalancePost).to.be.eq(pntRewardsManagerBalancePre + amount)
+    expect(await rewardsManager.depositedAmountByEpoch(0)).to.be.eq(amount)
+  })
+
+  it('should not be able to register for rewards without voting', async () => {
```

Comment:
Seeing what the test does, I think there's a mistake:
```suggestion
  it('should not be able to claim for rewards without voting', async () => {
```

Reply:
Indeed, you can't register if you didn't vote. As a consequence, you won't be able to claim as you haven't been registered.

Reply:
I think the phrase can be misinterpreted since I initially got that someone (randomguy?) can't register (hence call `registerRewardsForEpoch`) without voting. 
Probably worth clarifying a bit the statement with
```suggestion
  it('should not be able to claim for rewards without being registered', async () => {
```

Reply:
I just rephrased the test description as the test is more general.

Reply:
Yeah looks better now